src/xlsb_funs.h

```diff
@@ -188,13 +207,13 @@ std::string read_xlwidestring(std::string &mystring, std::istream& sas) {
 
   size_t size = mystring.size();
   std::u16string str;
-  str.resize((double)size * 2);
+  str.resize(size * 2);
 
   if (!sas.read((char*)&str[0], str.size()))
     Rcpp::stop("char: a binary read error occurred");
 
   std::string outstr = to_utf8(str);
-  if ((double)str.size()/2 != size) Rcpp::warning("String size unexpected");
+  if (str.size()/2 != size) Rcpp::warning("String size unexpected");
```

Comment:
A little bit unsure about this. Iirc I ran into an overflow issue here, but I'm unable to remember if this was caused by `size_t` or something else.